pkg/apis/options/cookie.go

```diff
@@ -27,7 +28,8 @@ func cookieFlagSet() *pflag.FlagSet {
 	flagSet.StringSlice("cookie-domain", []string{}, "Optional cookie domains to force cookies to (ie: `.yourcompany.com`). The longest domain matching the request's host will be used (or the shortest cookie domain if there is no match).")
 	flagSet.String("cookie-path", "/", "an optional cookie path to force cookies to (ie: /poc/)*")
 	flagSet.Duration("cookie-expire", time.Duration(168)*time.Hour, "expire timeframe for cookie")
-	flagSet.Duration("cookie-refresh", time.Duration(0), "refresh the cookie after this duration; 0 to disable")
+	flagSet.Duration("cookie-refresh", time.Duration(0), "refresh the cookie after this duration; 0 to disable; -1 to auto")
```

Comment:
I think we can set this it to be one or the other set, or neither set for no refreshing (and we just error in config validation if both are set). Makes the configuration easier for end users.

We default both to `0` and use whichever is non-zero.

Reply:
So you want duration to be eliminated altogether to remove TTL feature?

Reply:
I just meant no need to force a user to set `-1` to even enable the percentage based version.  Could just set the `cookie-refresh` was default `0` but `cookie-refresh-percent` was non-zero (ie 80).

But let me know your thoughts on this point: https://github.com/oauth2-proxy/oauth2-proxy/pull/886#discussion_r516455795